Add support for python3

JaviMerino opened the pull request:
Add support for python3, so that the ack-results-parser can be used with python2 or python3

python-hwinfo can't be used from pypi because it is python2 only.  This package depends on this pull request in `python-hwinfo`: https://github.com/rdobson/python-hwinfo/pull/25 , which adds python3 support to `python-hwinfo`.

Commit:
sfttp, xsautowf, xscertparser, xsjira: convert to python3

The result of running:

  futurize -w sfftp xsautowf xscertparser xsjira

Commit:
acklogparser: port to python-hwinfo 0.1.7

system info is now exported as a string rather than printing to the
screen.

stephenchengCloud commented on xscertparser/cmd/acklogparser.py:
```diff
@@ -61,29 +63,29 @@ def result_parser(tarfilename, logsubdir):  # pylint: disable=R0914,R0912
                     device.attributes['PCI_description'].value
                     )
-        if 'device' in device.attributes.keys():
+        if 'device' in list(device.attributes.keys()):
```
It's not related to your change to python3.
Not sure, but this line seems weird.
Shouldn't it be ` if 'id' in list(device.attributes.keys()): `

JaviMerino replied:
Indeed it is weird. What's worse is that it does not do anything! `hba_bus_id_list` is initialized in line 70 and appended to here, but the list is never used.  I have looked back at the history of the project and it has never been used since it was added in 2563a0b879d0 (back then it was called `hbaBusIdList`). 
I have removed the dead code.

stephenchengCloud commented on xscertparser/cmd/acklogparser.py:
```diff
@@ -61,29 +63,29 @@ def result_parser(tarfilename, logsubdir):  # pylint: disable=R0914,R0912
     # XS-version
     for version in test_conf.getElementsByTagName("global_config"):
-        if 'xs_version' in version.attributes.keys():
+        if 'xs_version' in list(version.attributes.keys()):
```
Here `list` is not necessary.
You can directly use ` if 'xs_version' in version.attributes: `

JaviMerino replied:
You are right, I have fixed this and all the other similar instances. I can't test it now, I'll test it tomorrow

JaviMerino replied:
Tested it this morning, works fine.

Commit:
acklogparser: remove hba_bus_id_list array

It's an array that has been appended to since it was introduced in
2563a0b879d0 and has never been used.  Remove the dead code.

stephenchengCloud commented:
> This package depends on this pull request in python-hwinfo: https://github.com/rdobson/python-hwinfo/pull/25 , which adds python3 support to python-hwinfo.

By the way, the fomal repo of python-hwinfo is https://github.com/xenserver/python-hwinfo, and it's already in Python3.

JaviMerino commented:
> By the way, the fomal repo of python-hwinfo is https://github.com/xenserver/python-hwinfo, and it's already in Python3.

I was fetching python-hwinfo from the wrong place. Thanks for letting me know.
